File: Assignment10/Assignment10Java/src/main/java/com/rishav/assignment/entity/Company.java
package com.rishav.assignment.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.time.LocalDate;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Company {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String regNo;

    private String Name;

    private LocalDate foundationDate;

    private List<Department> departments;

}

File: Assignment10/Assignment10TypeScript/src/model/Company.ts
import { Department } from './Department';
import { ICompany } from '../interfaces/ICompany' 

export class Company implements ICompany {
    
    collegeName: string;

    collegeFoundationDate: Date;

    private _regNo: string;

    private _departments: Array<Department> ;

    public get regNo() {
        return this._regNo;
    }

    public set regNo(regNo: string) {
        this._regNo = regNo;
    }

    public get departments() {
        return this._departments;
    }

    public set departments(departments: Array<Department>) {
        this._departments = departments;
    } 


}